lib/tasks/journeys_report.rake

```diff
@@ -1,65 +1,17 @@
 require 'csv'
 require 'json'
-namespace :journeys do
-  desc 'Exports a CSV report of all journeys'
-  task csv_report: :environment do
-    puts 'exporting journeys_report.csv...'
-
-    CSV.open('journeys_report.csv', 'w') do |csv|
-      csv << ['SUPPLIER', 'DATE', 'MOVE REFERENCE', 'MOVE FROM', 'MOVE TO', 'MOVE EVENTS', 'TIMESTAMP', 'JOURNEY FROM (title)', 'JOURNEY FROM (nomis id)', 'JOURNEY TO (title)', 'JOURNEY TO (nomis id)', 'JOURNEY EVENTS', 'VEHICLE', 'BILLABLE']
-
-      current_reference = nil
-      Journey
-          .joins(:move).includes(:move)
-          .joins(:supplier).includes(:supplier)
-          .joins(:from_location).includes(:from_location)
-          .joins(:to_location).includes(:to_location)
-          .order('suppliers.key ASC, moves.date ASC, moves.reference ASC, journeys.client_timestamp ASC')
-          .each do |journey| # NB: cannot do find_each because we need to order
-        csv << if current_reference != journey.move.reference
-                 # new move
-                 [journey.supplier.key,
-                  journey.move.date,
-                  journey.move.reference,
-                  journey.move.from_location.title,
-                  journey.move.to_location&.title,
-                  journey.move.move_events.map { |event| "#{event.event_name}: #{event.notes}" }.join('; '),
-                  journey.client_timestamp.strftime('%d/%m %H:%M:%S'),
-                  journey.from_location.title,
-                  journey.from_location.nomis_agency_id,
-                  journey.to_location.title,
-                  journey.to_location.nomis_agency_id,
-                  journey.events.map { |event| "#{event.event_name}: #{event.notes}" }.join('; '),
-                  journey.vehicle['registration'],
-                  journey.billable]
-               else
-                 # journey within current move
-                 [nil,
-                  nil,
-                  nil,
-                  nil,
-                  nil,
-                  nil,
-                  journey.client_timestamp.strftime('%d/%m %H:%M:%S'),
-                  journey.from_location.title,
-                  journey.from_location.nomis_agency_id,
-                  journey.to_location.title,
-                  journey.to_location.nomis_agency_id,
-                  journey.events.map { |event| "#{event.event_name}: #{event.notes}" }.join('; '),
-                  journey.vehicle['registration'],
-                  journey.billable]
-               end
-        current_reference = journey.move.reference
-      end
-    end
-  end
+require 'aws-sdk-s3'
 
+namespace :journeys do
   desc 'Exports a JSON report of all journeys'
   task json_report: :environment do
-    puts 'exporting journeys_report.json...'
-
+    timestamp = DateTime.now.utc.strftime('%s')
     suppliers = []
```

Comment:
`suppliers` probably isn't needed anymore as we are exporting one file per supplier: could you remove this variable and updates lines 63-69?

Reply:
I was thinking the same, but then I realised that the file can eventually be just imported without any context and the data inside it needs to have some way of telling which supplier the data belongs to without looking at the filename.

Reply:
ah yes, that is a good point. However, the way it is currently written the second supplier will include the data of the first supplier, so at the very least the `suppliers` variable needs to be re-initialised on every loop, and ideally renamed to `supplier`?

Reply:
Oh yes, that's a good point. Will fix that.

Reply:
Moved the `supplier` key inside the moves, so that each move object is tagged with the supplier it belongs to.